pkg/mutation/system.go

```diff
@@ -107,26 +115,59 @@ func (s *System) Mutate(obj *unstructured.Unstructured, ns *corev1.Namespace) (b
 			}
 			if cmp.Equal(original, obj) {
 				if *MutationLoggingEnabled {
-					logAppliedMutations("Oscillating mutation.", original, allAppliedMutations)
+					logAppliedMutations("Oscillating mutation.", mutationUUID, original, allAppliedMutations, nil)
 				}
 				return false, fmt.Errorf("oscillating mutation for %s %s %s %s", obj.GroupVersionKind().Group, obj.GroupVersionKind().Kind, obj.GetNamespace(), obj.GetName())
 			}
+			mutationTimestamp := time.Now()
 			if *MutationLoggingEnabled {
-				logAppliedMutations("Mutation applied", original, allAppliedMutations)
+				logAppliedMutations("Mutation applied", mutationUUID, original, allAppliedMutations, &mutationTimestamp)
+			}
+
+			if *MutationAnnotationsEnabled {
+				err := mutationAnnotations(obj, allAppliedMutations, mutationUUID)
+				if err != nil {
+					log.Error(err, "Error applying mutation annotations")
+				}
 			}
 			return true, nil
 		}
-		if *MutationLoggingEnabled {
+		if *MutationLoggingEnabled || *MutationAnnotationsEnabled {
 			allAppliedMutations = append(allAppliedMutations, appliedMutations)
 		}
 	}
 	if *MutationLoggingEnabled {
-		logAppliedMutations("Mutation not converging", original, allAppliedMutations)
+		logAppliedMutations("Mutation not converging", mutationUUID, original, allAppliedMutations, nil)
 	}
 	return false, fmt.Errorf("mutation not converging for %s %s %s %s", obj.GroupVersionKind().Group, obj.GroupVersionKind().Kind, obj.GetNamespace(), obj.GetName())
 }
 
-func logAppliedMutations(message string, obj *unstructured.Unstructured, allAppliedMutations [][]types.Mutator) {
+func mutationAnnotations(obj *unstructured.Unstructured, allAppliedMutations [][]types.Mutator, mutationUUID uuid.UUID) error {
+	mutatorStringSet := make(map[string]bool)
```

Comment:
nit: I think an empty struct (`struct{}`) uses zero bytes of storage, if we aren't planning on using the bool for anything

Reply:
Done